### DISCUSSION TITLE
Name completion for codespaces and other cloud coding environments

### DISCUSSION THREAD

**Discussion by mramsden** [blinksh/blink#1922] [Ideas]
I have a number of projects that I’m working on in Github and they all have individual codespaces. It would be great if the `code` command could offer tab completion on my available cloud coding environments.

My current workflow relies on remembering how the different codespaces are named (which Github autogenerates without a reference to the attached repository) and then using the command history to go and find the appropriate URL. This has the drawbacks that this list is removed with the command history as well as Github’s default naming scheme not being helpful.

Connecting the Blink app to the [Github API](https://docs.github.com/en/rest/codespaces/codespaces?apiVersion=2022-11-28#list-codespaces-for-the-authenticated-user) would allow an auto-discovery process of available coding environments that could offer a similar tab-completion style flow that exists with both the `ssh` and `mosh` commands.

In theory I can see how this feature might be relevant to gitpods and code-server instances as well although this might need further integration with the saved server connections feature as from what I can tell there is not a documented mechanism for listing available environments.

**Comment by carloscabanero** [comment]
Hi! Just to ack this. Yep, I have an idea for how to make it. Only issue is we are bit short in resources. 

So opening it here to the community, if someone has some availability and wants to do a (paid) short but very spectacular stunt in Blink, let me know.
